src/Symfony/Component/Console/Helper/ProgressIndicator.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class ProgressIndicator
+{
+    private $output;
+    private $startTime;
+    private $format;
+    private $message = null;
+    private $indicatorValues = array('-', '\\', '|', '/');
+    private $indicatorCurrent = 0;
+    private $lastMessagesLength = 0;
+
+    private static $formatters;
+    private static $formats;
+
+    /**
+     * @param OutputInterface $output
+     */
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->startTime = time();
+
+        $this->setFormat($this->determineBestFormat());
+    }
+
+    /**
+     * @param string $format
+     */
+    public function setFormat($format)
+    {
+        $this->format = self::getFormatDefinition($format);
+    }
+
+    /**
+     * @param string|null $message
+     */
+    public function setMessage($message)
+    {
+        $this->message = $message;
+
+        $this->display();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMessage()
+    {
+        return $this->message;
+    }
+
+    /**
+     * @param array $values
+     */
+    public function setIndicatorValues(array $values)
+    {
+        $values = array_values($values);
+
+        if (0 === count($values)) {
+            throw new \InvalidArgumentException('Must have at least 1 value.');
+        }
+
+        $this->indicatorValues = $values;
+    }
+
+    /**
+     * Gets the progress bar start time.
+     *
+     * @return int The progress bar start time
+     */
+    public function getStartTime()
+    {
+        return $this->startTime;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrentValue()
+    {
+        return $this->indicatorValues[$this->indicatorCurrent];
+    }
+
+    public function start($message)
+    {
+        $this->message = $message;
+        $this->startTime = time();
+        $this->indicatorCurrent = 0;
+
+        $this->display();
+    }
+
+    public function advance()
+    {
+        if ($this->indicatorCurrent === (count($this->indicatorValues) - 1)) {
+            $this->indicatorCurrent = 0;
+        } else {
+            ++$this->indicatorCurrent;
+        }
```

Comment:
I would just do a simple incrementation here (`++$this->indicatorCurrent;`), and then use a modulo where it is used.